lighthouse-core/test/lib/csp-evaluator-test.js

```diff
@@ -0,0 +1,105 @@
+ */
+'use strict';
+
+const {isIcuMessage} = require('../../lib/i18n/i18n.js');
+const {Type} = require('../../../third-party/csp-evaluator/optimized_binary.js');
+const {
+  evaluateRawCspForFailures,
+  getTranslatedDescription,
+} = require('../../lib/csp-evaluator.js');
+
+/* eslint-env jest */
+
+describe('Evaluator compatibility', () => {
+  it('finding types', () => {
+    expect(Type).toMatchInlineSnapshot(`
+      Object {
+        "100": "MISSING_SEMICOLON",
+        "101": "UNKNOWN_DIRECTIVE",
+        "102": "INVALID_KEYWORD",
+        "300": "MISSING_DIRECTIVES",
+        "301": "SCRIPT_UNSAFE_INLINE",
+        "302": "SCRIPT_UNSAFE_EVAL",
+        "303": "PLAIN_URL_SCHEMES",
+        "304": "PLAIN_WILDCARD",
+        "305": "SCRIPT_WHITELIST_BYPASS",
+        "306": "OBJECT_WHITELIST_BYPASS",
+        "307": "NONCE_LENGTH",
+        "308": "IP_SOURCE",
+        "309": "DEPRECATED_DIRECTIVE",
+        "310": "SRC_HTTP",
+        "400": "STRICT_DYNAMIC",
+        "401": "STRICT_DYNAMIC_NOT_STANDALONE",
+        "402": "NONCE_HASH",
+        "403": "UNSAFE_INLINE_FALLBACK",
+        "404": "WHITELIST_FALLBACK",
+        "405": "IGNORED",
+        "500": "REQUIRE_TRUSTED_TYPES_FOR_SCRIPTS",
+        "600": "REPORTING_DESTINATION_MISSING",
+        "601": "REPORT_TO_ONLY",
+        "DEPRECATED_DIRECTIVE": 309,
+        "IGNORED": 405,
+        "INVALID_KEYWORD": 102,
+        "IP_SOURCE": 308,
+        "MISSING_DIRECTIVES": 300,
+        "MISSING_SEMICOLON": 100,
+        "NONCE_HASH": 402,
+        "NONCE_LENGTH": 307,
+        "OBJECT_WHITELIST_BYPASS": 306,
+        "PLAIN_URL_SCHEMES": 303,
+        "PLAIN_WILDCARD": 304,
+        "REPORTING_DESTINATION_MISSING": 600,
+        "REPORT_TO_ONLY": 601,
+        "REQUIRE_TRUSTED_TYPES_FOR_SCRIPTS": 500,
+        "SCRIPT_UNSAFE_EVAL": 302,
+        "SCRIPT_UNSAFE_INLINE": 301,
+        "SCRIPT_WHITELIST_BYPASS": 305,
+        "SRC_HTTP": 310,
+        "STRICT_DYNAMIC": 400,
+        "STRICT_DYNAMIC_NOT_STANDALONE": 401,
+        "UNKNOWN_DIRECTIVE": 101,
+        "UNSAFE_INLINE_FALLBACK": 403,
+        "WHITELIST_FALLBACK": 404,
+      }
+    `);
+  });
+  it.todo('descriptions');
+});
+
+describe('getTranslatedDescription', () => {
+  // Missing directives
+  it('missing script-src', () => {
+    const rawCsp = `base-uri 'none'; object-src 'none'`;
+    const findings = evaluateRawCspForFailures([rawCsp]);
+    const translated = findings.map(getTranslatedDescription);
+
+    expect(translated).toHaveLength(1);
+    expect(isIcuMessage(translated[0])).toBeTruthy();
+    expect(translated[0]).toBeDisplayString(
+      'script-src directive is missing. This can allow the execution of unsafe scripts.'
+    );
+  });
```

Comment:
Remaining tests would look the same as this one.

Reply:
Are these tests already done in the library?

Reply:
Yes, these tests are focused on testing translation.